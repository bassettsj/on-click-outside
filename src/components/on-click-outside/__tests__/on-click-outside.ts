import {
  OnClickOutsideElement
} from '../on-click-outside';
import { TestWindow } from '@stencil/core/testing';

describe('On Click Outside element', () => {
  let el: HTMLElement;
  let win: TestWindow;
  beforeEach(async () => {
    win = new TestWindow();
    el = await win.load({
      html: '<on-click-outside><p>hi</p></on-click-outside>',
      components: [OnClickOutsideElement],
    });
  });
  test('should be able to render', async () => {
    expect(el).toMatchSnapshot();
  });
  describe('Handle Body Click listener', () => {
    test('it should emit a custom clickoutside event when click targent doesn\'t contain the element', (done) => {
      win.document.querySelector('on-click-outside').addEventListener('clickoutside', (event) => {
        expect(event.type).toBe('clickoutside');
        expect(
          // @ts-ignore
          event.detail
        ).toBeDefined();
        done();
      });
      win.document.body.click();
    });
    test('it should not emit a custom clickoutside event when click target is in the element ', (done) => {
      win.document.querySelector('on-click-outside').addEventListener('clickoutside', (event) => {
        done('Failed!');
      });
      win.document.addEventListener('click', () => {
        setTimeout(() => {
          done();
        }, 0);
      });
      win.document.querySelector('p').click();
    });
  });
});
