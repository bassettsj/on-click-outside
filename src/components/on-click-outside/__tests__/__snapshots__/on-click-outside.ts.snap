// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`On Click Outside element should be able to render 1`] = `
<on-click-outside
  class="hydrated"
>
  <p>
    hi
  </p>
</on-click-outside>
`;
